In [1]:
import pandas as pd
import numpy as np

In [3]:
#loading dataset

df = pd.read_csv(r"C:\Users\User\Desktop\week 1&2\1_E-Commerce_Dataset_Cleaning&_EDA\OnlineRetail.csv", encoding="latin" )

In [4]:
# Display the first few rows

df.head()

,InvoiceNo,StockCode,Description,Quantity,InvoiceDate,UnitPrice,CustomerID,Country
0,536365,85123A,WHITE HANGING HEART T-LIGHT HOLDER,6,12/1/2010 8:26,2.55,17850.0,United Kingdom
1,536365,71053,WHITE METAL LANTERN,6,12/1/2010 8:26,3.39,17850.0,United Kingdom
2,536365,84406B,CREAM CUPID HEARTS COAT HANGER,8,12/1/2010 8:26,2.75,17850.0,United Kingdom
3,536365,84029G,KNITTED UNION FLAG HOT WATER BOTTLE,6,12/1/2010 8:26,3.39,17850.0,United Kingdom
4,536365,84029E,RED WOOLLY HOTTIE WHITE HEART.,6,12/1/2010 8:26,3.39,17850.0,United Kingdom


In [5]:
# Number of rows and columns

df.shape

(541909, 8)

In [6]:
# Data type of all columns

df.info()

<class 'pandas.DataFrame'>
RangeIndex: 541909 entries, 0 to 541908
Data columns (total 8 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   InvoiceNo    541909 non-null  str    
 1   StockCode    541909 non-null  str    
 2   Description  540455 non-null  str    
 3   Quantity     541909 non-null  int64  
 4   InvoiceDate  541909 non-null  str    
 5   UnitPrice    541909 non-null  float64
 6   CustomerID   406829 non-null  float64
 7   Country      541909 non-null  str    
dtypes: float64(2), int64(1), str(5)
memory usage: 67.3 MB


In [7]:
# Numeric features

df.select_dtypes(include= ["int64", "float64"])

,Quantity,UnitPrice,CustomerID
0,6,2.55,17850.0
1,6,3.39,17850.0
2,8,2.75,17850.0
3,6,3.39,17850.0
4,6,3.39,17850.0
...,...,...,...
541904,12,0.85,12680.0
541905,6,2.10,12680.0
541906,4,4.15,12680.0
541907,4,4.15,12680.0


In [8]:
# categorical features

df.select_dtypes(include = "object")

C:\Users\User\AppData\Local\Temp\ipykernel_8528\3190643171.py:3: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  df.select_dtypes(include = "object")


,InvoiceNo,StockCode,Description,InvoiceDate,Country
0,536365,85123A,WHITE HANGING HEART T-LIGHT HOLDER,12/1/2010 8:26,United Kingdom
1,536365,71053,WHITE METAL LANTERN,12/1/2010 8:26,United Kingdom
2,536365,84406B,CREAM CUPID HEARTS COAT HANGER,12/1/2010 8:26,United Kingdom
3,536365,84029G,KNITTED UNION FLAG HOT WATER BOTTLE,12/1/2010 8:26,United Kingdom
4,536365,84029E,RED WOOLLY HOTTIE WHITE HEART.,12/1/2010 8:26,United Kingdom
...,...,...,...,...,...
541904,581587,22613,PACK OF 20 SPACEBOY NAPKINS,12/9/2011 12:50,France
541905,581587,22899,CHILDREN'S APRON DOLLY GIRL,12/9/2011 12:50,France
541906,581587,23254,CHILDRENS CUTLERY DOLLY GIRL,12/9/2011 12:50,France
541907,581587,23255,CHILDRENS CUTLERY CIRCUS PARADE,12/9/2011 12:50,France


In [13]:
# Possible unique identifers

round((df.nunique()/df.count())*100,)

InvoiceNo      5.0
StockCode      1.0
Description    1.0
Quantity       0.0
InvoiceDate    4.0
UnitPrice      0.0
CustomerID     1.0
Country        0.0
dtype: float64

### Brief Description

1. The dataset contains transaction records from an online retail store. Each row represents a single product purchased within an invoice, with multiple rows sharing the same invoice number when a customer purchases more than one item in a transaction.

2. The dataset contains `541909` rows and `8` columns.

3. The dataset does not contain any column that act as the unique identifiers.

4. The dataset has 2 columns stored with the wrong data type **InvoiceDate** and 
**CustomerID**.


